tests/test_datasets/test_cams.py

```diff
@@ -0,0 +1,130 @@
+"""Unit test for CAMS dataset."""
+
+import json
+from pathlib import Path
+from unittest.mock import patch
+import numpy as np
+import pytest
+import xarray as xr
+from zampy.datasets.catalog import CAMS
+from zampy.datasets.dataset_protocol import SpatialBounds
+from zampy.datasets.dataset_protocol import TimeBounds
+from . import data_folder
+
+
+@pytest.fixture(scope="function")
+def valid_path_config(tmp_path_factory):
+    """Create a dummy .zampy_config file."""
+    fn = tmp_path_factory.mktemp("usrhome") / "zampy_config.yml"
+    with open(fn, mode="w", encoding="utf-8") as f:
+        f.write("cdsapi:\n  url: a\n  key: 123:abc-def\n")
+        f.write("adsapi:\n  url: a\n  key: 123:abc-def")
+    return fn
+
+
+@pytest.fixture(scope="function")
+def dummy_dir(tmp_path_factory):
+    """Create a dummpy directory for testing."""
+    return tmp_path_factory.mktemp("data")
+
+
+class TestCAMS:
+    """Test the CAMS class."""
+
+    @patch("cdsapi.Client.retrieve")
+    def test_download(self, mock_retrieve, valid_path_config, dummy_dir):
+        """Test download functionality.
+        Here we mock the downloading and save property file to a fake path.
+        """
+        times = TimeBounds(np.datetime64("2003-01-02"), np.datetime64("2020-12-31"))
+        bbox = SpatialBounds(54, 56, 1, 3)
+        variable = ["co2_concentration"]
+        cds_var_names = ["carbon_dioxide"]
+        download_dir = Path(dummy_dir, "download")
+
+        cams_dataset = CAMS()
+        # create a dummy .cdsapirc
+        patching = patch("zampy.datasets.cds_utils.CONFIG_PATH", valid_path_config)
+        with patching:
+            cams_dataset.download(
+                download_dir=download_dir,
+                time_bounds=times,
+                spatial_bounds=bbox,
+                variable_names=variable,
+                overwrite=True,
+            )
+
+            # make sure that the download is called
+            mock_retrieve.assert_called_once_with(
+                "cams-global-ghg-reanalysis-egg4",
+                {
+                    "model_level": "60",
+                    "variable": cds_var_names,
+                    "date": f"{str(times.start)}/{str(times.end)}",
+                    "step": ["0", "3", "6", "9", "12", "15", "18", "21"],
```

Comment:
I couldn't find where to set `model_level` and `step` when requesting data? :thinking:

Reply:
They are hardcoded in `cds_utils.py`, since they are fixed for running stemmus-scope.